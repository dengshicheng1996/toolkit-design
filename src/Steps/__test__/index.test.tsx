import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Steps } from '..'

const { Step } = Steps

describe('Steps', () => {
  it('horizontal snapshot', () => {
    const component = shallow(
      <Steps current={3} direction="horizontal">
        <Step title="Finished" data-y="1" />
        <Step title="In Progress" description="This is description" />
        <Step title="Waiting" description="This is description" />
      </Steps>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  it('vertical snapshot', () => {
    const component = shallow(
      <Steps current={2} direction="vertical">
        <Step title="Finished" description="This is description" />
        <Step title="In Progress" description="This is description" />
        <Step title="Waiting" />
        <Step title="Final" description="This is description" />
      </Steps>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })
})
