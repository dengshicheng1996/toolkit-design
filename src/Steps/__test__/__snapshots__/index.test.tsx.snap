// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps horizontal snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "current": 3,
      "direction": "horizontal",
      "length": 3,
    }
  }
>
  <div
    className="amd-steps horizontal"
  >
    <Step
      data-y="1"
      index={0}
      key=".0"
      title="Finished"
    />
    <Step
      description="This is description"
      index={1}
      key=".1"
      title="In Progress"
    />
    <Step
      description="This is description"
      index={2}
      key=".2"
      title="Waiting"
    />
  </div>
</ContextProvider>
`;

exports[`Steps vertical snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "current": 2,
      "direction": "vertical",
      "length": 4,
    }
  }
>
  <div
    className="amd-steps vertical"
  >
    <Step
      description="This is description"
      index={0}
      key=".0"
      title="Finished"
    />
    <Step
      description="This is description"
      index={1}
      key=".1"
      title="In Progress"
    />
    <Step
      index={2}
      key=".2"
      title="Waiting"
    />
    <Step
      description="This is description"
      index={3}
      key=".3"
      title="Final"
    />
  </div>
</ContextProvider>
`;
